cpp/core/jni/JniCommon.h

```diff
@@ -264,22 +265,27 @@ class CelebornClient : public RssClient {
       return;
     }
     env->DeleteGlobalRef(javaCelebornShuffleWriter_);
+    env->DeleteGlobalRef(array_);
   }
 
   int32_t pushPartitonData(int32_t partitionId, char* bytes, int64_t size) {
     JNIEnv* env;
     if (vm_->GetEnv(reinterpret_cast<void**>(&env), jniVersion) != JNI_OK) {
       throw gluten::GlutenException("JNIEnv was not attached to current thread");
     }
-    jbyteArray array = env->NewByteArray(size);
-    env->SetByteArrayRegion(array, 0, size, reinterpret_cast<jbyte*>(bytes));
+    jint length = env->GetArrayLength(array_);
+    if (size > length) {
+      array_ = env->NewByteArray(size);
```

Comment:
shall we release the array_ first then assign the new size ?

Reply:
done